### PR TITLE
added error message to endchatfailed

### PR DESCRIPTION
**Pull request by mithunjohny**
# PR Details

### Description

_Include a description of the problem to be solved_

## Solution Proposed

EndChat exceptionDetails missing the error message from the oc-sdk sessionclose api call
Also need to split request in promiseAll because eve if sessionInit fails ,other promises where continuing in background causing ACS connection created without a successfull sesionInit

### Acceptance criteria

_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence

_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests

- [ ] You have tested all changes in Popout mode
- [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
- [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y

- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

***Please provide justification if any of the validations has been skipped.***

### CODE REVIEW

File: src/OmnichannelChatSDK.ts [hunk 1]
```diff
@@ -968,6 +968,9 @@ class OmnichannelChatSDK {
                 RequestId: this.requestId,
                 ChatId: this.chatToken.chatId as string
             };
+            if (error instanceof ChatSDKError) {
```
Comment (xTEddie) [inline src/OmnichannelChatSDK.ts:971]: Is this error specific to `persistent chat`?

Comment (mithunjohny) [reply src/OmnichannelChatSDK.ts:971]: No , it is to catch exception rethrown from closeChat method in this file.

**Review by xTEddie — APPROVED** [review]
